File: Latihan3-cpp/PC.cpp
#include "Processor.cpp"
#include "Disk.cpp"
#include "RAM.cpp"

class PC{

    private:

    Processor p;
    Disk d;
    RAM r;
    int totalPrice;

    public:

    // Constructor
    PC(){
    }

    PC(Processor p, Disk d, RAM r, int totalPrice){
        this->p = p;
        this->d = d;
        this->r = r;
        this->totalPrice = p.getPrice() + d.getPrice() + r.getPrice();
    }

    // Get-set Processor
    void setProcessor(Processor p){
        this->p = p;
    }

    Processor getProcessor(){
        return this->p;
    }

    // Get-set Disk
    void setDisk(Disk d){
        this->d = d;
    }

    Disk getDisk(){
        return this->d;
    }

    // Get-set RAM
    void setRAM(RAM r){
        this->r = r;
    }

    RAM getRAM(){
        return this->r;
    }

    // Get-set totalPrice
    void setTotalPrice(int p1, int p2, int p3){
        this->totalPrice = p1 + p2 + p3;
    }

    int getTotalPrice(){
        return this->totalPrice;
    }


    // Deconstructor
    ~PC(){
    }

};

File: Latihan3-py/PC.py
from Processor import Processor
from Disk import Disk
from RAM import RAM

class PC:
    # Constructor
    def __init__(self):
        self.__p = Processor()
        self.__d = Disk()
        self.__r = RAM()
        self.__totalPrice = 0
    
    # Get-set Processor
    def setProcessor(self, p):
        self.__p = p

    def getProcessor(self):
        return self.__p

    # Get-set Disk
    def setDisk(self, d):
        self.__d = d

    def getDisk(self):
        return self.__d
    
    # Get-set RAM
    def setRAM(self, r):
        self.__r = r

    def getRAM(self):
        return self.__r

    # Get-set totalPrice
    def setTotalPrice(self, p1, p2, p3):
        self.__totalPrice = p1 + p2 + p3

    def getTotalPrice(self):
        return self.__totalPrice